tap_gitlab/tests/test_core.py

```diff
@@ -43,3 +48,28 @@ def test_tap_config_defaults():
     tap = TapGitLab(config=SAMPLE_CONFIG, parse_env_config=True)
     for optin_stream in OPTIN_STREAM_NAMES:
         assert f"fetch_{optin_stream}" in tap.config
+
+
+def test_pkeys_in_schema():
+    """Verify that primary keys are actually in the stream's schema."""
+    tap = TapGitLab(config=SAMPLE_CONFIG, parse_env_config=True)
+    for name, stream in tap.streams.items():
+        pkeys = stream.primary_keys or []
+        schema_props = set(stream.schema["properties"].keys())
+        for pkey in pkeys:
+            assert (
+                pkey in schema_props
+            ), f"Coding error in stream {name}: primary_key {pkey} is missing in schema"
```

Comment:
This was a big recurring bug in my testing. Thanks for adding this test.

Reply:
Would it make sense to add such checks in the sdk itself? I don't know how well it applies to other types of taps, say ones with more dynamic schema, but it seems to me that a tap should not be able to pass tests/CI checks and still return invalid records.

I can open a PR in the `sdk` if you think it makes sense.

Reply:
Closing this as the test PR in the sdk was merged, and I just upgraded the sdk version here.